src/tools/jsondocck/src/main.rs

```diff
@@ -205,7 +205,21 @@ fn check_command(command: Command, cache: &mut Cache) -> Result<(), CkError> {
                     let val = cache.get_value(&command.args[0])?;
                     let results = select(&val, &command.args[1]).unwrap();
                     let pat = string_to_value(&command.args[2], cache);
-                    results.contains(&pat.as_ref())
+                    let has = results.contains(&pat.as_ref());
+                    // Give better error for when @has doesnt conatian
+                    if !command.negated && !has {
+                        return Err(CkError::FailedCheck(
+                            format!(
+                                "{} matched to {:?} but didnt have {:?}",
```

Comment:
Eventually it would be nice to walk up the pattern until you find a match, the way error-reporting for intra-doc links works, but it will be a lot of work and doesn't need to happen here.
```suggestion
                                "{} matched to {:?} but didn't have {:?}",
```
https://github.com/rust-lang/rust/blob/861872bc453bde79b83ff99d443d035225f10e87/src/librustdoc/passes/collect_intra_doc_links.rs#L1785-L1786